### DISCUSSION TITLE
Give SDK Runner the ability to temporarily store data

### DISCUSSION THREAD

**Discussion by colonelchlorine** [Geotab/sdk#295] [Ideas]
In the SDK Runner, it would be useful to have some type of session or window state available that is retained between "Run" actions. As the code runs in an iFrame that only has access to call, multiCall, console.log, console.error - it can't use localStorage, sessionStorage and any objects added to `window` are destroyed when "Run" is pressed.

Stack received when executing `sessionStorage.setItem('abc', 123);
```
Uncaught DOMException: Failed to read the 'sessionStorage' property from 'Window': The document is sandboxed and lacks the 'allow-same-origin' flag.
    at window.___injectScript___ (<anonymous>:3:1)
    at injectJavaScript (https://geotab.github.io/sdk/software/api/codebase/iframe.html:137:77)
    at Object.injectSample [as successCallback] (https://geotab.github.io/sdk/software/api/codebase/iframe.html:119:44)
    at gotPostMessage (https://geotab.github.io/sdk/software/api/codebase/iframe.html:70:38)
    at filterMessages (https://geotab.github.io/sdk/software/api/codebase/iframe.html:59:25)
```

https://github.com/Geotab/sdk/blob/master/src/software/api/codebase/iframe.html#L100

Ideas:
- A new iFrame message that enables setting & getting a value in the parent window
- Give the iFrame more control of the parent window with the `allow-same-origin` flag

Both have risks, but so far my experimenting hasn't found any temporary storage hack, so it's not possible to cache results so each run doesn't make excessive API calls. Idea came from trying to build an SDK tool that is URL accessible & stores data in sessionStorage.

**Comment by DawsonMyers** [comment]
I've actually already been working on something that does just that (adds new messages for communicating with the iframe). Started it last month, but got busy with other things and haven't had much extra time to work on it.

**Comment by colonelchlorine** [comment]
@DawsonMyers Had another thought just now .. it would be cool if the URL hash could have arbitrary parameters (similar to the "save" one currently there). Was just thinking it would be neat to save the URL and send it to someone with form state populated.
